add the `voxpopuli` recipe

vesis84 opened the pull request:
- this is the data preparation
- there is no ASR training and no results

vesis84 commented:
Hi, who would review this ?
K.

csukuangfj commented on egs/voxpopuli/ASR/README.md:
```diff
@@ -0,0 +1,36 @@
+# Readme
+
+This recipe contains data preparation for the VoxPopuli dataset.
```
Could you provide a hyperlink for VoxPopuli?
E.g., use something like below
```
[VoxPopuli](http://some/ulr/for/VoxPopuli)
```

csukuangfj commented on egs/voxpopuli/ASR/local/compute_fbank.py:
```diff
@@ -0,0 +1,242 @@
+and CutSet manifest stored in `data/fbank/{prefix}_cuts_{dataset}.jsonl.gz`.
+
+The number of workers is smaller than nunber of jobs
+(see --num-jobs 100 --num-workers 25 in the example).
```
Is it a hard constraint? If so, could you write a check in the code for it?

vesis84 replied:
it is not a hard constraint, there can be also less jobs  than workers, in this case some workers will be idle...

vesis84 replied:
ok, the assert is added into the code

csukuangfj commented on egs/voxpopuli/ASR/local/preprocess_voxpopuli.py:
```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3
+# Copyright    2023  Xiaomi Corp.        (authors: Yifan Yang)
```
Maybe you need to update the author information?

vesis84 replied:
ok, done here and also in other .py scripts...

csukuangfj commented on egs/voxpopuli/ASR/local/separate_punctuation.py:
```diff
@@ -0,0 +1,131 @@
+#!/bin/env python3
```
```suggestion
#!/usr/bin/env python3
```

csukuangfj commented on egs/voxpopuli/ASR/local/separate_punctuation.py:
```diff
@@ -0,0 +1,131 @@
+
+Usage example #2:
+  python3 ./local/separate_punctuation.py \
+    --ignore-columnts 1 \
```
```suggestion
    --ignore-columns 1 \
```

csukuangfj commented on egs/voxpopuli/ASR/local/separate_punctuation.py:
```diff
@@ -0,0 +1,131 @@
+Usage example #2:
+  python3 ./local/separate_punctuation.py \
+    --ignore-columnts 1 \
+    ${kaldi_data}/text
```
This code seems to read the content from `stdin`, not the provided file `${kaldi_data}/text`. Is it correct?

csukuangfj commented on egs/voxpopuli/ASR/local/text_from_manifest.py:
```diff
@@ -0,0 +1,48 @@
+#!/bin/env python3
```
```suggestion
#!/usr/bin/env python3
```

csukuangfj commented on egs/voxpopuli/ASR/local/uppercase_begin_of_sentence.py:
```diff
@@ -0,0 +1,114 @@
+#!/bin/env python3
```
```suggestion
#!/usr/bin/env python3
```

csukuangfj commented on egs/voxpopuli/ASR/local/uppercase_begin_of_sentence.py:
```diff
@@ -0,0 +1,114 @@
+
+Usage example #2:
+  python3 ./local/uppercase_begin_of_sentence.py \
+    --ignore-columnts 1 \
```
```suggestion
    --ignore-columns 1 \
```

csukuangfj commented on egs/voxpopuli/ASR/local/validate_cutset_manifest.py:
```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
```
Could you replace this file with a symlink to its original one if no changes are done to this file?

vesis84 replied:
Actually, it is extended version of `validate_manifest.py` from Librispeech. Should I change it directly in the Librispeech recipe ? (for now i'll update it in the VoxPopuli recipe, and i'll add validation to `prepare.sh` too)

csukuangfj commented on egs/voxpopuli/ASR/prepare.sh:
```diff
@@ -0,0 +1,263 @@
+#!/usr/bin/env bash
+
+. /mnt/matylda5/iveselyk/ASR_TOOLKITS/K2_SHERPA_PYTORCH20/conda-activate.sh
```
Please add the following two lines
https://github.com/k2-fsa/icefall/blob/6d275ddf9fdd67a32b79b93d70fedffe4b156d5c/egs/librispeech/ASR/prepare.sh#L3-L4

csukuangfj commented on egs/voxpopuli/ASR/prepare.sh:
```diff
@@ -0,0 +1,263 @@
+# "sk", "sl", "et", "lt", "pt", "bg", "el", "lv", "mt", "sv", "da",
+# "asr", "10k", "100k", "400k"
+#
+# See: https://github.com/lhotse-speech/lhotse/blob/c5f26afd100885b86e4244eeb33ca1986f3fa923/lhotse/bin/modes/recipes/voxpopuli.py#L77
```
This URL links to
```
@click.option(
    "--subset",
    type=click.Choice(["asr", "10k", "100k", "400k"] + LANGUAGES + LANGUAGES_V2),
    default="asr",
)
```

Should it be associated with the next line
```
task=asr
```
?

csukuangfj commented on egs/voxpopuli/ASR/prepare.sh:
```diff
@@ -0,0 +1,263 @@
+fi
+
+if [ $stage -le 5 ] && [ $stop_stage -ge 5 ]; then
+  log "Stage 6: Compute fbank for train set of VoxPopuli"
```
```suggestion
  log "Stage 5: Compute fbank for train set of VoxPopuli"
```

csukuangfj commented:
> Hi, who would review this ? K.

Sorry for the late reply.

Just left some minor comments. Otherwise, everything looks good to me.

vesis84 commented:
Hi @csukuangfj , thank you for the review. The issues should be fixed now.
I'll re-run the data preparation to see if it's okay.
K.

csukuangfj commented:
> Hi @csukuangfj , thank you for the review. The issues should be fixed now. I'll re-run the data preparation to see if it's okay. K.

Thanks!

Looks good to me. Please leave a message when you think it is ready to merge.

vesis84 commented:
Hi @csukuangfj ,
to me it seems done. I fixed or explained all the issues, fixed quite some typos too. 
I did a re-run of the data preparation here locally.

Please, have a look one more time, if possible.
Best regards,
Karel

csukuangfj commented:
@vesis84 

Thanks for your contribution!
